internal/process/process.go

```diff
@@ -28,10 +28,13 @@ func ActivationPID(cfg Configurable) int32 {
 	procInfoErrMsgFmt := "Could not detect process information: %v"
 
 	for pid != 0 && pid != ppid {
+		logging.Debug("Current PID: %d, Parent PID: %d", pid, ppid)
 		pidFileName := ActivationPIDFileName(cfg.ConfigPath(), int(pid))
+		logging.Debug("Looking for activation pid file: %s", pidFileName)
 		if fileutils.FileExists(pidFileName) {
 			return pid
 		}
+		logging.Debug("Activation pid file not found")
```

Comment:
I'm having trouble wrapping my head around this function. We return "successfully" early on Line 35 _before_ initializing the process and attempting to get the PID, so all that verification below seems pointless. I could add a name check and debug there but that wouldn't be for the case we are trying to verify. Initializing the process earlier makes this function more error-prone but we could ignore those errors.

Reply:
I don't think it would produce the same results since it is updating pid and ppid, then looping. The -1 returns collapse the loop before the inspection finds itself back to process zero, but it all does seem reasonably useful.

Reply:
I see what you mean. I've added a name debug but I'm not sure how helpful it will be.